tools/cldr-apps/js/src/views/ReportResponse.vue

```diff
@@ -90,60 +87,77 @@ export default {
           this.acceptable = false;
           break;
       }
-      const theUrl = `api/voting/reports/users/${user.id}/locales/${this.$cldrOpts.locale}/reports/${this.report}`;
-      await cldrAjax
-        .doFetch(theUrl, {
-          method: "POST",
-          body: JSON.stringify({
-            acceptable: this.acceptable,
-            completed: this.completed,
-          }),
-          headers: {
-            "Content-Type": "application/json",
-            "X-SurveyTool-Session": this.$cldrOpts.sessionId,
-          },
-        })
-        .catch((e) => {
-          console.error(e);
-          this.error = e;
-          this.loaded = true;
-        });
-      return this.reload(); // will set loaded=true
     },
-    reload() {
+    /**
+     * map 2 fields to a string
+     */
+    getFields(fields) {
+      const { completed, acceptable } = fields;
+      if (acceptable && completed) {
+        return "acceptable";
+      } else if (!acceptable && completed) {
+        return "unacceptable";
+      } else {
+        return "incomplete";
+      }
+    },
+    async changed() {
       this.loaded = false;
       const user = cldrStatus.getSurveyUser();
       if (!user) {
         this.error = "Not logged in.";
         this.loaded = true;
         return;
       }
-      const theUrl = `api/voting/reports/users/${user.id}/locales/${this.$cldrOpts.locale}`;
-      return cldrAjax
-        .doFetch(theUrl, {
-          headers: {
-            "X-SurveyTool-Session": this.$cldrOpts.sessionId,
+      try {
+        this.setFields(this.state);
+        await this.client.apis.voting.updateReport(
```

Comment:
@btangmu  is that shorter? This function `voting.updateReport()` is synthesized by the API definition the server side

Reply:
really amazing -- i need to study this new ability to use swagger on the front end!